File: Solutions/6kyu/6kyu_divisor_harmony.py
def solve(a,b):
    pairs={}
    for i in range(a,b):
        ratio=div_sum(i)/i
        try:    pairs[ratio]=pairs[ratio]+[i]
        except: pairs[ratio]=[i]
    return sum(min(i) for i in pairs.values() if len(i)>=2)

    
def div_sum(n):
    return sum(i for i in range(1,n+1) if n%i==0)

File: Solutions/6kyu/6kyu_divisor_harmony.js
function solve(a,b){
  var pairs = {}
  for (var a; a<b; a++){
    const ratio=divSum(a)/a
    pairs[ratio]==undefined ? pairs[ratio]=[a] : pairs[ratio].push(a)
  }
  return Object
         .values(pairs)
         .filter(e=>e.length>=2)
         .reduce((acc,val)=> acc+Math.min(...val),0)
         
};


function divSum(n){
  var total = 0
  for (var i=1; i<n+1; i++) n%i==0 ? total+=i : undefined
  return total
}
